### PR TITLE
Fix sarch and add some catalan translations

### PR DESCRIPTION
**Pull request by Xevib**
- Try to fix https://github.com/organicmaps/organicmaps/issues/9179 
- Add some missing translations in catalan

### CODE REVIEW

File: data/categories.txt [hunk 1]
```diff
@@ -2298,6 +2303,7 @@ el:Αγορά|αγορά|κατάστημα
 sk:3Trhovisko|samoobsluha
 sw:Soko
 fa:بازار
+ca:Mercat
```
Comment (biodranik) [inline data/categories.txt:2306]: Thanks! Can you pleas try to sort added translations in all places?

Comment (Xevib) [reply data/categories.txt]: Thank you for the advice, I ordered all ca strings

File: data/categories.txt [hunk 2]
```diff
@@ -2299,6 +2323,7 @@ sk:3Trhovisko|samoobsluha
 sw:Soko
 fa:بازار
 
+
```
Comment (biodranik) [inline data/categories.txt:2326]:
```suggestion
```

File: data/categories.txt [hunk 3]
```diff
@@ -2389,6 +2416,7 @@ ru:Автодом|Продажа автодомов
 tr:Karavan galerisi
 uk:Продаж автобудинків
 
+
```
Comment (biodranik) [inline data/categories.txt:2419]:
```suggestion
```

**Review by biodranik — APPROVED** [review]
Thanks!

**Pull request merged**